docs/src/notes.md

```diff
@@ -0,0 +1,28 @@
+# Notes
+
+## Using Solvers and Other Additional Packages
+Certain solvers may require a license or an additional installation. Since we want to test the tutorials using Travis CI and run them in the browser using Binder, we should always try to go for the solver which works out of the box with just a Package install. In case this is not possible, do not add the tutorial to the `runtests.jl` file and add a line that it does not work with Binder. Note that this issue is not specific to solvers and may arise with other Julia packages as well.
+
+## Tangle
+We use the tangle feature of Weave.jl to generate files for testing as it allows us extract code from a file. This enables us to skip cetains blocks of code we would not want for tests. These include-
+### Deliberate Errors
+Tutorials might contain examples of what not to do and hence some code blocks can throw errors on purpose. These blocks cause tests to fail when we run the entire tutorial and hence they are skipped from tangle.
+### Installing Packages
+THough code that installs packages will not cause tests to fail, it will slow them down. Since we are already adding all the required packages as dependencies, blocks that install packages are skipped from tangle. 
+
```

Comment:
I would put spaces before and after every header line.

Also, if you are listed out things and you might want to consider using `**bold text here**` instead of `### bold text here`.